### DISCUSSION TITLE
Intercept HTTP connection close with Quarkus RESTeasy Reactive

### DISCUSSION THREAD

**Discussion by ppatierno** [quarkusio/quarkus#30597] [Q&A] [answered]
When using an HTTP server in Vert.x (then the Routes to set endpoints), I am also able to set a `connectionHandler` so that with the provided `HttpConnection` instance I can intercept when it's closed by the client and doing some stuff on my side.
I am not able to find a similar way to do that by using Quarkus RESTeasy Reactive. Am I missing anything?

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
We don't really have something similar in RESTEasy Reactive. 

What's the use case you are trying to cover?

**Reply by ppatierno** [reply]
Our Strimzi HTTP to Kafka bridge is moving to Quarkus instead of using Vert.x (directly).
When an HTTP client wants to produce a message to Kafka, the bridge creates a Kafka producer instance that needs to be keep if the connection is "keep-alive" from the HTTP client but it has to be closed when the HTTP clients drop the connection (to avoid leaving a Kafka producer instance still active but not used on the bridge). This HTTP client connection to Kafka producer (on the bridge) will be also useful when we'll need to enable a passthrough authentication from HTTP to Kafka. So a 1:1 HTTP client - Kafka producer relationship is needed.
Isn't there anything else (i.e. HTTP session?) that allows me to map the HTTP client with a corresponding Kafka producer? Today our bridge just takes a map of HttpConnection to producer.

**Reply by geoand** [reply]
We don't have anything OOTB (but we probably should provide something). 
But it sounds very much like you can do something similar to what is mentioned in: https://github.com/quarkusio/quarkus/issues/30250#issue-1524997421.

WDYT?

**Reply by ppatierno** [reply]
If I got the workaround right, it helps to commit the transaction when the response from server to client ends. I think it helps the user (in the specific example) to avoid to commit first but then something bad happens when sending the response which doesn't arrive to the client. So in his use case, when we are sure the response is consumed by the client (in the `bodyEndHandler`), the transaction is closed.

I think my use case is slightly different, because the end of the HTTP response doesn't mean that the client will even close the underneath TCP connection right after that. It could have the connection "keep-alive" so I cannot use the `bodyEndHandler` to close my Kafka producer (because if the HTTP client sends a new HTTP request on the same connection, the same Kafka producer has to be used).

**Reply by geoand** [reply]
You could get the `HttpConnection` by using `routingContext.request().connection()` and working with that. It's not elegant, but maybe worth a first try to see if it fits your needs?

**Reply by ppatierno** [reply]
Yeah I was looking at it but the problem is that the `RoutingContext` is actually part of the vertx-web component and not the vertx-core. We are getting rid of any other Vertx dependency which is not strictly part of Quarkus. So I guess everything in vertx-core is fine, but not vertx-web.

**Reply by geoand** [reply]
RESTEasy Reactive depends on Vert.x Web, so you will have that anyway if you move to it :)

**Reply by ppatierno** [reply]
Ok then! I will try with the suggested workaround, thanks!

**Reply by geoand** [reply]
👍🏼 

Let us know how it goes

**Reply by ppatierno** [reply]
I will do. As per our offline discussion, here the issue I opened for this use case. Thanks!

https://github.com/quarkusio/quarkus/issues/30599
